Sources/InkCLI/main.swift

```diff
@@ -7,23 +7,60 @@
 import Foundation
 import Ink
 
-guard CommandLine.arguments.count > 1 else {
-    print("""
-    Ink: Markdown -> HTML converter
-    -------------------------------
-    Pass a Markdown string to convert as input,
-    and HTML will be returned as output. To use
-    STDIN as input, call ink with "-" as a single
-    argument, like this: '$ ink -'.
-    """)
+if CommandLine.arguments.contains(where: { $0 == "-h" || $0 == "--help" }) {
+    print(helpMessage, on: .standardOut)
     exit(0)
 }
 
-var markdown = CommandLine.arguments[1]
+if CommandLine.arguments.contains(where: { $0 == "--version" }) {
+    print(versionMessage, on: .standardOut)
+    exit(0)
+}
+
+let markdown: String
 
-if markdown == "-" {
+if CommandLine.arguments.count == 1 {
```

Comment:
I think this could be implemented in a very clean way using pattern matching:

```swift
let arguments = CommandLine.arguments

switch arguments.count {
case 1:
    // parse stdin
case 2:
    // parse file
case 3 where arguments[1] == "-m" || arguments[1] == "--markdown":
    // parse inline markdown
default:
    // error handling
}
```

Reply:
Much cleaner. I left the -m flag check as the second case, so that if you run `ink -m` or `ink -m string1 string2` you get an error message specific to the -m flag.

Reply:
Yeah the new implementation looks fantastic 👍